crates/kornia-imgproc/src/core.rs

```diff
@@ -106,15 +105,24 @@ pub fn bitwise_and<const CHANNELS: usize>(
     src1: &Image<u8, CHANNELS>,
     src2: &Image<u8, CHANNELS>,
     mask: &Image<u8, 1>,
-) -> Result<Image<u8, CHANNELS>> {
-    assert!(
-        src1.size() == src2.size(),
-        "The input images must have the same size",
-    );
-    assert!(
-        src1.size() == mask.size(),
-        "The input images and the mask must have the same size"
-    );
+) -> Result<Image<u8, CHANNELS>, ImageError> {
+    if src1.size() != src2.size() {
+        return Err(ImageError::InvalidImageSize(
```

Comment:
why not just pass the two image sources?

Reply:
i prefer to keep the errors with primitive types to avoid import conflicts in the future